### PR TITLE
Add a lazy loaded content to the local resource

### PR DESCRIPTION
**Pull request by msm-code**
Support for retreiving content from resources without an explicit content load:

```python
        config_res = Resource(name="config.json", path=config_path)
        config = json.loads(config_res.content.decode())
```

This has a nice side-effect of typing `.content` property more strictly (it becomes `bytes` instead of `Optional[bytes]`)

### CODE REVIEW

File: karton/core/resource.py [hunk 1]
```diff
@@ -201,6 +201,19 @@ def __init__(
...
 
+    @property
+    def content(self) -> bytes:
+        """
+        Resource content. Reads the file if the file was not read before.
+
+        :return: Content bytes
+        """
+        if self._content is None:
+            assert self._path is not None
```
Comment (nazywam) [inline karton/core/resource.py:212]: We can either get the content as `content` or from `path`, right? There is no other way

**Review by nazywam — APPROVED** [review]

File: karton/core/resource.py (hunk 1 above)
Comment (psrok1) [reply karton/core/resource.py:212]: We can check additionally in LocalResource constructor if at least one thing (content/path) is provided and raise ValueError instead.

File: karton/core/resource.py [hunk 2]
```diff
@@ -106,13 +106,13 @@ def uid(self) -> str:
...
-    def content(self) -> Optional[bytes]:
+    def content(self) -> bytes:
         """
-        Resource content
+        Resource content. Must be overriden in a derived class.
 
         :return: Resource contents
         """
-        return self._content
+        raise NotImplementedError()
```
Comment (psrok1) [inline karton/core/resource.py:115]: :+1: Yup, it's better to keep it abstract on this level

**Review by psrok1 — APPROVED** [review]
Looks ok, left comment on assertion

**Review by psrok1 — APPROVED** [review]

**Pull request merged**